### DISCUSSION TITLE
How to simulate Fourier Transform at the Focal Plane (when input_distance = f)

### DISCUSSION THREAD

**Discussion by udithhaputhanthri** [rafael-fuente/Diffraction-Simulations--Angular-Spectrum-Method#29] [Q&A] [answered]
Hi, 
Thank you for your fantastic work.

I have implemented the lens system as given in [this script](https://github.com/rafael-fuente/diffractsim/blob/main/examples/optical_imaging_system.py). I also followed [your article](https://rafael-fuente.github.io/simulating-light-diffraction-with-lenses-visualizing-fourier-optics.html#references) as well.

When I set up input_distance = output_distance = focal_length= 25cm, Ideally I should get the Fourier Transform at the output. But the implementation does not give that. Could you please guide me on how I can get that?

Input field (intensity)
![image](https://user-images.githubusercontent.com/53050875/186714150-18c42ce8-028c-403e-a2bb-1f96547c306c.png)

What I got from the simulation (I checked for multiple scale factors in scale_propagation function (in [this script](https://github.com/rafael-fuente/diffractsim/blob/main/diffractsim/propagation_methods/two_steps_fresnel_method.py)). Below are images for scale_factor= 30)
![image](https://user-images.githubusercontent.com/53050875/186714204-174bd17f-6ff3-4abf-bc98-11549ddedd61.png)

What should I get (Fourier Transform)
![image](https://user-images.githubusercontent.com/53050875/186714316-45d0238c-2f98-42fd-bc4b-64a9132c2279.png)
![image](https://user-images.githubusercontent.com/53050875/186714453-e6517f1e-5257-4926-ba75-7ec2709dc271.png)

**Comment by rafael-fuente** [comment, marked as answer]
Hi, 

I edited the script to accommodate the parameters you mentioned (input_distance = output_distance = focal_length= 25cm)

```python
import diffractsim
diffractsim.set_backend("CPU") #Change the string to "CUDA" to use GPU acceleration

from diffractsim import MonochromaticField, ApertureFromImage, Lens, nm, mm, cm, um


zi = 25*cm # distance from the image plane to the lens
z0 = 25*cm # distance from the lens to the current position
radius = 6*mm


# set up simulation
F = MonochromaticField(
    wavelength=488 * nm, extent_x=1.5 * mm, extent_y=1.5 * mm, Nx=2048, Ny=2048,intensity = 0.4
)

F.add(ApertureFromImage("./apertures/QWT.png",  image_size = (1.0 * mm, 1.0 * mm), simulation = F))

F.scale_propagate(25*cm, scale_factor = 20)
F.add(Lens(f = 25*cm , radius = radius))
F.scale_propagate(25*cm, scale_factor = 0.7)

I = F.get_intensity()
F.plot_intensity(I, square_root = True, units = mm, grid = True)

```
This is the output:
![Figure_1](https://user-images.githubusercontent.com/37953893/188296527-98f72f97-d28c-49f8-a9f7-5b35286c4436.png)


That is the absolute value of the Fourier transform of the aperture, at least within the paraxial approximation.
